python_modules/dagster/dagster/_core/definitions/assets_job.py

```diff
@@ -85,13 +82,13 @@ def get_base_asset_jobs(
                 *asset_graph.asset_keys_for_partitions_def(partitions_def=partitions_def),
                 *asset_graph.unpartitioned_asset_keys,
             }
-            loadable_asset_keys = asset_graph.all_asset_keys - executable_asset_keys
             jobs.append(
                 build_assets_job(
                     f"{ASSET_BASE_JOB_PREFIX}_{i}",
-                    executable_assets=asset_graph.assets_defs_for_keys(executable_asset_keys),
-                    loadable_assets=asset_graph.assets_defs_for_keys(loadable_asset_keys),
-                    asset_checks=asset_graph.asset_checks_defs,
+                    # For now, to preserve behavior keep all asset checks in all base jobs.
```

Comment:
Can you further elaborate here? Why is the ideal behavior? Why is this "For now"?

Reply:
Checks should probably just be in the base job with their target asset, and I'm almost certain that will need to be the case when they support partitions. I've added a comment to that effect.